fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/joinorder/hypergraph/HyperGraph.java

```diff
@@ -304,6 +304,76 @@ private long calNodeMap(Set<Slot> slots) {
         return bitmap;
     }
 
+    public static HyperGraph from(Plan plan) {
+        Preconditions.checkArgument(plan.getGroupExpression().isPresent(),
```

Comment:
Should support plan without group. May be inside the methed from(Plan plan), call the memo.init(), and then build graph